File: 0101_0200/0165/0165.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m = map(int, f.readline().split())
    k = [int(x) for x in f.read().split()]
    p = [0] * (m * n)
    p[0] = 1
    for i in range(n * m - 1):
        if p[i] and i % m + k[i] < m:
            p[k[i] + i] += p[i]
        if p[i] and i / m + k[i] < n:
            p[k[i] * m + i] += p[i]
    q.write(str(p[-1]))

File: 0101_0200/0165/0165.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, m, t int
	fmt.Fscan(f, &n, &m)
	p := make([]int, n*m)
	p[0] = 1
	for i := 0; i < n*m-1; i++ {
		fmt.Fscan(f, &t)
		if p[i] > 0 && t+i%m < m {
			p[t+i] += p[i]
		}
		if p[i] > 0 && t+i/m < n {
			p[t*m+i] += p[i]
		}
	}
	fmt.Fprint(q, p[n*m-1])
}
